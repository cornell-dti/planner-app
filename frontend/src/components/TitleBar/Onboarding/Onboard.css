.Hero{
  position:fixed;
  top:0;
  left:0;
  width:100%;
  height:100%;
  overflow-y:auto;
  background:#7fbdff;
  z-index:10;
  color:white;
}

.SettingsSection {
  margin-top: 10px;
}

p.SettingsSectionTitle {
  vertical-align: top;
  font-size: 16px;
  padding-top: 20px;
}

.SettingsSectionContent {
  vertical-align: top;
}

@media only screen and (min-width: 768px) {
  p.SettingsSectionTitle {
    width: 200px;
    display: inline-block;
  }

  .SettingsSectionContent {
    display: inline-block;
    width: calc(100% - 200px);
  }
}

.SettingsButton button {
  color: black;
  border: 1px white solid;
  border-radius: 5px;
  padding: 5px 10px;
  line-height: 14px;

  margin-top: 5px;
  cursor:pointer;
}

.SignButton {
  background: white !important;
  border: none !important;
  border-radius: 20px !important;
  font-size: 16px !important;
  padding: 10px 20px !important;
  
  cursor:pointer;
}

.ColorConfigItemList {
  display: inline-block;
  width: 100%;
  margin: 0;
  padding: 0;
  list-style: none;
}

.TutorialImg{
  width:100%;
  height:100%;
  max-height:100vh;
  object-fit:contain;
  object-position:center center;
}
