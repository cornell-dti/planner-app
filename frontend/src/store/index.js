// @flow strict

import { createStore } from 'redux';
import type { Store } from 'redux';
import rootReducer from './reducers';
import type { State } from './store-types';
import type { Action } from './action-types';

const store: Store<State, $Subtype<Action>> = createStore(rootReducer);
export default store;
