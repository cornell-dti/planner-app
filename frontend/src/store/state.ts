import { Map, Set } from 'immutable';
import { State } from 'common/lib/types/store-types';
import { NONE_TAG_ID, NONE_TAG } from 'common/lib/util/tag-util';

/**
 * The initial state of the app.
 * This state is dummy. It needs to be patched by the data from the backend ASAP.
 * @type {State}
 */
export const initialState: State = {
  tags: Map({ [NONE_TAG_ID]: NONE_TAG }),
  tasks: Map(),
  missingSubTasks: Map(),
  orphanSubTasks: Map(),
  dateTaskMap: Map(),
  groupTaskMap: Map(),
  repeatedTaskSet: Set(),
  groupTaskSet: Set(),
  settings: { canvasCalendar: null, completedOnboarding: true, theme: 'light' },
  bannerMessageStatus: {},
  courses: Map(),
  pendingInvites: Map(),
  groups: Map(),
};

/**
 * The initial state setup for testing.
 * It contains some random values so that snapshot tests can be more meaningful.
 * @type {State}
 */
export const initialStateForTesting: State = {
  tags: Map({
    [NONE_TAG_ID]: NONE_TAG,
    foo: { id: 'foo', order: 1, name: 'Foo', color: 'red', classId: null },
    bar: { id: 'bar', order: 2, name: 'Bar', color: 'blue', classId: null },
    baz: { id: 'baz', order: 3, name: 'Baz: Baz', color: 'green', classId: '2112' },
  }),
  tasks: Map({
    foo: {
      id: 'foo',
      order: 1,
      owner: 'foo',
      name: 'Foo',
      tag: 'foo',
      complete: true,
      inFocus: false,
      children: [{ id: 'foo', order: 1, name: 'Foo', complete: true, inFocus: false }],
      metadata: {
        type: 'ONE_TIME',
        date: new Date('2030-01-01'),
      },
    },
    bar: {
      id: 'bar',
      order: 2,
      owner: 'foo',
      name: 'Bar',
      tag: 'bar',
      complete: false,
      inFocus: true,
      children: [{ id: 'bar', order: 2, name: 'Bar', complete: false, inFocus: true }],
      metadata: {
        type: 'ONE_TIME',
        date: new Date('2030-01-01'),
      },
    },
    baz: {
      id: 'baz',
      order: 3,
      owner: 'foo',
      name: 'Baz',
      tag: 'baz',
      complete: true,
      inFocus: true,
      children: [{ id: 'baz', order: 3, name: 'Baz', complete: true, inFocus: true }],
      metadata: {
        type: 'ONE_TIME',
        date: new Date('2030-01-01'),
      },
    },
  }),
  dateTaskMap: Map({
    [new Date('2030-01-01').toDateString()]: Set(['foo']),
    [new Date('2031-01-01').toDateString()]: Set(['bar', 'baz']),
    [new Date('2032-01-01').toDateString()]: Set(),
  }),
  groupTaskMap: Map({
    groupId1: Set(['foo']),
    groupId2: Set(['bar', 'baz']),
    groupId3: Set(),
  }),
  missingSubTasks: Map(),
  orphanSubTasks: Map(),
  repeatedTaskSet: Set(),
  groupTaskSet: Set(),
  settings: { canvasCalendar: null, completedOnboarding: true, theme: 'light' },
  bannerMessageStatus: {},
  courses: Map({
    CS2112: [
      {
        courseId: 42,
        subject: 'CS',
        courseNumber: '2112',
        title: 'OO Design Data Structs Honors',
        examTimes: [
          { type: 'final', time: 1576332000000 },
          { type: 'prelim', time: 1570145400000 },
        ],
      },
    ],
  }),
  pendingInvites: Map(),
  groups: Map(),
};
