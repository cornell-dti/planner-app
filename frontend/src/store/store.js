// @flow strict

import { createStore } from 'redux';
import type { Store } from 'redux';
import rootReducer from './reducers';
import type { State } from './store-types';
import type { Action } from './action-types';
import type { AppUser } from '../util/firebase-util';
import { error } from '../util/general-util';

export type GlobalStore = Store<State, $Subtype<Action>>;

const store: GlobalStore = createStore(rootReducer);
let appUser: AppUser | null = null;

/**
 * Initialize the app and record a global user and returns the initialized store.
 *
 * @param {AppUser} user the user of the app.
 * @return {GlobalStore} the newly initialized store.
 */
export function initializeApp(user: AppUser): GlobalStore {
  appUser = user;
  return store;
}

/**
 * Returns the global app user.
 *
 * @return {AppUser} the global app user.
 */
export const getAppUser = (): AppUser => appUser ?? error('App is not initialized.');

/**
 * Dispatch an action and returns the dispatched action.
 *
 * @param {Action} action the action to dispatch.
 * @return {Action} the action to dispatch.
 */
export const dispatchAction = (action: Action): Action => store.dispatch(action);
