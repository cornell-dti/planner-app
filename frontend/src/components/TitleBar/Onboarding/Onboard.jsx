// @flow strict

import * as React from 'react';
import type { Node } from 'react';
import TagItem from '../Tags/TagItem';
import ClassTagAdder from '../Tags/ClassTagAdder';
import styles from './Onboard.css';
import type { Tag, Task } from '../../../store/store-types';
import { importCourseExams } from '../../../store/actions';
import { dispatchConnect, stateConnect } from '../../../store/react-redux-util';

/**
 * The class adder component.
 *
 * @return {Node} rendered component.
 * @constructor
 */
const ClassAdder = (): Node => (
  <div className={styles.SettingsSection}>
    <div className={styles.SettingsSectionContent}>
      <ClassTagAdder />
    </div>
  </div>
);

/**
 * The exam exporter component.
 *
 * @return {Node} rendered component.
 * @constructor
 */
const ExamImporter = dispatchConnect({ onClick: importCourseExams })(({ onClick }) => (
  <button type="button" onClick={onClick} className={styles.SignButton}>Import Exams</button>
));

/**
 * The tags container component.
 *
 * @param {string} title the title of the tags.
 * @param {Node} children the tags as children.
 * @return {Node} rendered component.
 * @constructor
 */
const TagsContainer = ({ title, children }: {| +title: string; +children: Node |}): Node => (
  <div className={styles.SettingsSection}>
    <p className={styles.SettingsSectionTitle}>{title}</p>
    <div className={styles.SettingsSectionContent}>
      <ul className={styles.ColorConfigItemList}>
        {children}
      </ul>
    </div>
  </div>
);

type Props = {| +tags: Tag[]; +tasks: Task[]; |};
type State = {| +progress: number; +shouldDisp: boolean; |};

/**
 * The settings page.
 *
 * @param {Tag[]} tags all the tags.
 * @return {Node} the rendered editor.
 * @constructor
 */
class Onboard extends React.PureComponent<Props, State> {
  state: State = { progress: 0, shouldDisp: false };

  showNext = () => this.setState((state: State) => ({ ...state, progress: state.progress + 1 }));

  goBack = () => this.setState((state: State) => {
    const { progress } = state;
    if (progress > 1) {
      return { progress: progress - 1 };
    }
    return {};
  });

  skipTutorial = () => this.setState(state => ({ ...state, progress: 100 }));

  render() {
    const classTags: Tag[] = [];
    const otherTags: Tag[] = [];
    const { tags, tasks } = this.props;
    const { shouldDisp, progress } = this.state;
    tags.forEach((tag) => {
      if (tag.classId !== null) {
        classTags.push(tag);
      } else if (tag.name !== 'None') {
        otherTags.push(tag);
      }
    });

    if (classTags.length === 0 && otherTags.length === 0 && tasks.length === 0) {
      // Using an if statement here b/c we want shouldDisp's value to persist
      // even after a couple of tags have been added
      this.setState(state => ({ ...state, shouldDisp: true }));
    }
    const renderTags = (arr: Tag[]): Node => arr.map((tag: Tag) => (
      <TagItem key={tag.id} tag={tag} />
    ));

    const importAll = (r) => {
      const images = {};
      r.keys().forEach((item) => { images[item.replace('./', '')] = r(item); });
      return images;
    };

    const images = importAll(require.context('../../../assets/tutorial', false, /\.(png|jpe?g|svg)$/));

    return (
      <div
        className={styles.Hero}
        style={{
          display: shouldDisp && progress < 7 ? 'block' : 'none',
          overflowY: progress > 0 ? 'hidden' : '',
          background: progress > 0 ? 'rgba(0,0,0,0.8)' : '',
        }}
      >
        <div style={{ display: progress === 0 ? 'block' : 'none', padding: '60px 40px' }}>
          <p className={styles.HeroIntroText}>
            Hi! Help us boost your productivity by creating some tags.
            <br />
            Search and add the classes you are currently enrolled in to tag them.
          </p>
          <ClassAdder />
          <h2>Class Tags</h2>
          <TagsContainer title="Class Tags">
            {renderTags(classTags)}
          </TagsContainer>
          <div>
            <ExamImporter />
            <button
              type="button"
              className={styles.SignButton}
              onClick={this.showNext}
              style={{ display: classTags.length !== 0 ? 'block' : 'none' }}
            >
              Done
            </button>
          </div>
        </div>
        <div style={{ display: progress > 0 ? 'block' : 'none' }} className={styles.ModalWrap}>
          <h2>Tutorial</h2>
          <div className={styles.TutorialModal}>
            <button type="button" onClick={this.goBack}>&lsaquo;</button>
            <img
              className={styles.TutorialImg}
              src={progress > 0 && progress < 7 ? images[`t${progress}.png`] : ''}
              alt="Tutorial"
            />
            <button type="button" onClick={this.showNext}>&rsaquo;</button>
          </div>
          <p className={styles.ModalOptions}>
            <button type="button" onClick={this.skipTutorial} style={{ marginRight: '30px' }}>
              Start using Samwise now &rarr;
            </button>
          </p>
        </div>
      </div>
    );
  }
}

const Connected = stateConnect<Props, Props>(
  ({ tags, tasks }) => ({ tags, tasks }),
)(Onboard);
export default Connected;
