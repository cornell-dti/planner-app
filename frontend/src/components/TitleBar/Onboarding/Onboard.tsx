import React, { ReactElement, CSSProperties, useState } from 'react';
import { connect } from 'react-redux';
import TagItem from '../Tags/TagItem';
import ClassTagAdder from '../Tags/ClassTagAdder';
import styles from './Onboard.css';
import { State, Tag } from '../../../store/store-types';
import Tutorial1 from '../../../assets/tutorial/t1.png';
import Tutorial2 from '../../../assets/tutorial/t2.png';
import Tutorial3 from '../../../assets/tutorial/t3.png';
import Tutorial4 from '../../../assets/tutorial/t4.png';
import Tutorial5 from '../../../assets/tutorial/t5.png';
import Tutorial6 from '../../../assets/tutorial/t6.png';
import { completeOnboarding, importCourseExams } from '../../../firebase/actions';

type TaggedImage = { readonly src: string; readonly alt: string }

const taggedImages: TaggedImage[] = [
  {
    src: Tutorial1,
    alt: 'Sam is your personal assistant.',
  },
  {
    src: Tutorial2,
    alt: 'Add tasks in the box \'What do you have to do?\'',
  },
  {
    src: Tutorial3,
    alt: 'Press the pin icon to move a task to focus view. Press the unpin icon to remove it.',
  },
  {
    src: Tutorial4,
    alt: 'Tasks added will go into future view',
  },
  {
    src: Tutorial5,
    alt: 'You can track the progress in the progress bar',
  },
  {
    src: Tutorial6,
    alt: 'You can review the tutorial and edit tags and classes in settings',
  },
];

type AddClassProps = { readonly classTags: Tag[]; readonly showNext: () => void };

/**
 * Adding class as the first onboarding step.
 */
function AddClassOnBoarding({ classTags, showNext }: AddClassProps): ReactElement {
  return (
    <div style={{ padding: '60px 40px' }}>
      {/* some random intro */}
      <p className={styles.HeroIntroText}>
        Hi! Help us boost your productivity by creating some tags.
        <br />
        Search and add the classes you are currently enrolled in to tag them.
      </p>
      {/* class adder */}
      <div className={styles.SettingsSection}>
        <div className={styles.SettingsSectionContent}>
          <ClassTagAdder />
        </div>
      </div>
      {/* display all class tags */}
      <h2>Class Tags</h2>
      <div className={styles.SettingsSection}>
        <p className={styles.SettingsSectionTitle}>Class Tags</p>
        <div className={styles.SettingsSectionContent}>
          <ul className={styles.ColorConfigItemList}>
            {classTags.map(tag => <TagItem key={tag.id} tag={tag} />)}
          </ul>
        </div>
      </div>
      {/* buttons */}
      <div className={styles.SignButtonContainer}>
        <button type="button" onClick={importCourseExams} className={styles.SignButton}>
          Import Exams
        </button>
        <span className={styles.Padding} />
        <button type="button" className={styles.SignButton} onClick={showNext}>
          {classTags.length > 0 ? 'Done' : 'Skip Adding Classes'}
        </button>
      </div>
    </div>
  );
}

type SlidesProps = {
  readonly progress: number;
  readonly showNext: () => void;
  readonly goBack: () => void;
  readonly skipTutorial: () => void;
};

/**
 * Showing slides as the second onboarding step.
 */
function SlidesOnBoarding({ progress, showNext, goBack, skipTutorial }: SlidesProps): ReactElement {
  const { src, alt } = taggedImages[progress - 1];
  return (
    <div className={styles.ModalWrap}>
      <h2>Tutorial</h2>
      <div className={styles.TutorialModal}>
        <button type="button" onClick={goBack}>&lsaquo;</button>
        <img className={styles.TutorialImg} src={src} alt={alt} />
        <button type="button" onClick={showNext}>&rsaquo;</button>
      </div>
      <p className={styles.ModalOptions}>
        <button type="button" onClick={skipTutorial} style={{ marginRight: '30px' }}>
          Start using Samwise now &rarr;
        </button>
      </p>
    </div>
  );
}

type Props = { readonly classTags: Tag[]; readonly completedOnboarding: boolean };

/**
 * The onboarding page. Displayed only if the user chooses to do so or during first landing.
 */
function Onboard({ classTags, completedOnboarding }: Props): ReactElement | null {
  const [progress, setProgress] = useState<number>(0);

  if (completedOnboarding || progress >= 7) {
    if (progress >= 7) {
      setProgress(0);
    }
    completeOnboarding(true);
    // the conditions not to display the tutorial
    return null;
  }

  const showNext = (): void => setProgress(prev => prev + 1);
  const goBack = (): void => setProgress(prev => (prev > 1 ? prev - 1 : prev));
  const skipTutorial = (): void => setProgress(100);

  const onboardingContainerStyle: CSSProperties | undefined = progress > 0
    ? { overflowY: 'hidden', background: 'rgba(0,0,0,0.8)' }
    : undefined;
  return (
    <div className={styles.Hero} style={onboardingContainerStyle}>
      {progress === 0 && <AddClassOnBoarding classTags={classTags} showNext={showNext} />}
      {progress > 0 && (
        <SlidesOnBoarding
          progress={progress}
          showNext={showNext}
          goBack={goBack}
          skipTutorial={skipTutorial}
        />
      )}
    </div>
  );
}

const Connected = connect(
  ({ tags, settings: { completedOnboarding } }: State): Props => {
    const classTags: Tag[] = Array.from(tags.values()).filter(t => t.classId != null);
    return { classTags, completedOnboarding };
  },
)(Onboard);
export default Connected;
