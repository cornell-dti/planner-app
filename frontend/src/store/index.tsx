import React, { ReactElement } from 'react';
import { Provider as ReactReduxProvider } from 'react-redux';
import { store, storeForTesting } from './store';

type Props = { readonly children: ReactElement };

export const Provider = ({ children }: Props): ReactElement => (
  <ReactReduxProvider store={store}>{children}</ReactReduxProvider>
);

export const ProviderForTesting = ({ children }: Props): ReactElement => (
  <ReactReduxProvider store={storeForTesting}>{children}</ReactReduxProvider>
);
