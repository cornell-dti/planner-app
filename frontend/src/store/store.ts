import { createStore, Store } from 'redux';
import rootReducer from './reducers';
import { State } from './store-types';
import { Action } from './action-types';
import { initialStateForTesting } from './state';

export type GlobalStore = Store<State, Action>;

/**
 * The redux store. It should only be given to the react-redux provider.
 */
export const store: GlobalStore = createStore(rootReducer);

/**
 * The redux store for testing.
 */
export const storeForTesting: GlobalStore = createStore(
  (initialState: State = initialStateForTesting, action: Action): State => rootReducer(
    initialState, action,
  ),
);

/**
 * Dispatch an action and returns the same dispatched action.
 */
export const dispatchAction = (action: Action): Action => store.dispatch(action);
