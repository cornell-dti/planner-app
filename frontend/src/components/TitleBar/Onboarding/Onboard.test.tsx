import React from 'react';
import renderer from 'react-test-renderer';
import { ProviderForTesting } from '../../../store';
import { initialStateForTesting } from '../../../store/state';
import { Onboard } from './Onboard';

it('TitleBar matches snapshot', () => {
  const tree = renderer
    .create(
      <ProviderForTesting>
        <Onboard
          classTags={Array.from(initialStateForTesting.tags.values())}
          completedOnboarding={false}
        />
      </ProviderForTesting>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
