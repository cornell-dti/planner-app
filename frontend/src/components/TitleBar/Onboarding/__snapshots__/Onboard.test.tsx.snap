// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleBar matches snapshot 1`] = `
<div
  className="Hero"
>
  <div
    style={
      Object {
        "padding": "60px 40px",
      }
    }
  >
    <p
      className="HeroIntroText"
    >
      Hi! Help us boost your productivity by creating some tags.
      <br />
      Search and add the classes you are currently enrolled in to tag them.
    </p>
    <div
      className="SettingsSection"
    >
      <div
        className="SettingsSectionContent"
      >
        <div
          className="TagColorConfigItemAdder SearchClasses"
          title="Search for a class"
        >
          <div>
            <input
              className="SearchInput"
              onChange={[Function]}
              placeholder="Search for classes (e.g. CS 2110, Introduction to Creative Writing)"
              type="text"
              value=""
            />
            <div />
          </div>
        </div>
      </div>
    </div>
    <h2>
      My Classes
    </h2>
    <div
      className="SettingsSection"
    >
      <div
        className="SettingsSectionContent"
      >
        <ul
          className="ColorConfigItemList"
        >
          <li
            className="ColorConfigItem"
          >
            <span
              className="TagName"
              style={
                Object {
                  "width": "calc(100% - 175px)",
                }
              }
            >
              <input
                className="TagEdit"
                onChange={[Function]}
                type="text"
                value="None"
              />
            </span>
            <button
              className="ColorEdit"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="ColorDisplay"
                style={
                  Object {
                    "backgroundColor": "#969696",
                  }
                }
              />
              <span
                title="More"
              >
                <svg
                  alt="More"
                  className="Arrow"
                  height="1em"
                  onKeyUp={[Function]}
                  tabIndex={0}
                  title="More"
                  width="1em"
                >
                  v.svg
                </svg>
              </span>
            </button>
            <button
              className="DeleteTag"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                title="Delete"
              >
                <svg
                  alt="Delete"
                  height="1em"
                  onKeyUp={[Function]}
                  tabIndex={0}
                  title="Delete"
                  width="1em"
                >
                  XLight.svg
                </svg>
              </span>
            </button>
          </li>
          <li
            className="ColorConfigItem"
          >
            <span
              className="TagName"
              style={
                Object {
                  "width": "calc(100% - 175px)",
                }
              }
            >
              <input
                className="TagEdit"
                onChange={[Function]}
                type="text"
                value="Foo"
              />
            </span>
            <button
              className="ColorEdit"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="ColorDisplay"
                style={
                  Object {
                    "backgroundColor": "red",
                  }
                }
              />
              <span
                title="More"
              >
                <svg
                  alt="More"
                  className="Arrow"
                  height="1em"
                  onKeyUp={[Function]}
                  tabIndex={0}
                  title="More"
                  width="1em"
                >
                  v.svg
                </svg>
              </span>
            </button>
            <button
              className="DeleteTag"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                title="Delete"
              >
                <svg
                  alt="Delete"
                  height="1em"
                  onKeyUp={[Function]}
                  tabIndex={0}
                  title="Delete"
                  width="1em"
                >
                  XLight.svg
                </svg>
              </span>
            </button>
          </li>
          <li
            className="ColorConfigItem"
          >
            <span
              className="TagName"
              style={
                Object {
                  "width": "calc(100% - 175px)",
                }
              }
            >
              <input
                className="TagEdit"
                onChange={[Function]}
                type="text"
                value="Bar"
              />
            </span>
            <button
              className="ColorEdit"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="ColorDisplay"
                style={
                  Object {
                    "backgroundColor": "blue",
                  }
                }
              />
              <span
                title="More"
              >
                <svg
                  alt="More"
                  className="Arrow"
                  height="1em"
                  onKeyUp={[Function]}
                  tabIndex={0}
                  title="More"
                  width="1em"
                >
                  v.svg
                </svg>
              </span>
            </button>
            <button
              className="DeleteTag"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                title="Delete"
              >
                <svg
                  alt="Delete"
                  height="1em"
                  onKeyUp={[Function]}
                  tabIndex={0}
                  title="Delete"
                  width="1em"
                >
                  XLight.svg
                </svg>
              </span>
            </button>
          </li>
          <li
            className="ColorConfigItem"
          >
            <span
              style={
                Object {
                  "display": "inline-block",
                  "width": "calc(100% - 175px)",
                }
              }
            >
              <span
                className="TagName"
              >
                Baz
              </span>
              <span
                className="ClassExpandedTitle"
              >
                Baz
              </span>
            </span>
            <button
              className="ColorEdit"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="ColorDisplay"
                style={
                  Object {
                    "backgroundColor": "green",
                  }
                }
              />
              <span
                title="More"
              >
                <svg
                  alt="More"
                  className="Arrow"
                  height="1em"
                  onKeyUp={[Function]}
                  tabIndex={0}
                  title="More"
                  width="1em"
                >
                  v.svg
                </svg>
              </span>
            </button>
            <button
              className="DeleteTag"
              onClick={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                title="Delete"
              >
                <svg
                  alt="Delete"
                  height="1em"
                  onKeyUp={[Function]}
                  tabIndex={0}
                  title="Delete"
                  width="1em"
                >
                  XLight.svg
                </svg>
              </span>
            </button>
          </li>
        </ul>
      </div>
    </div>
    <p
      className="ImportExamContainer"
    >
      <label
        htmlFor="onboardImportClass"
      >
        <input
          checked={true}
          id="onboardImportClass"
          onChange={[Function]}
          type="checkbox"
        />
        Yes, please automatically import registrar-scheduled exams for my classes.
      </label>
    </p>
    <div
      className="SignButtonContainer"
    >
      <span
        className="Padding"
      />
      <button
        className="SignButton"
        onClick={[Function]}
        type="button"
      >
        Done
      </button>
    </div>
  </div>
</div>
`;
